packages/fuels-signers/src/wallet.rs

```diff
@@ -560,9 +543,13 @@ impl WalletUnlocked {
         tx_parameters: TxParameters,
     ) -> Result<Vec<Receipt>> {
         let predicate = self.get_provider()?;
-        let spendable_predicate_resources = predicate
-            .get_spendable_resources(predicate_address, asset_id, amount)
-            .await?;
+
+        let filter = ResourceFilter {
+            from: predicate_address.clone(),
+            amount,
+            ..Default::default()
+        };
+        let spendable_predicate_resources = predicate.get_spendable_resources(filter).await?;
```

Comment:
Did we mean to use `provider` here ?
```suggestion
        let spendable_predicate_resources = provider.get_spendable_resources(filter).await?;
```